### DISCUSSION TITLE
Hydreon RG-15 Rain Sensor configuration

### DISCUSSION THREAD

**Discussion by minniR2D2** [arendst/Tasmota#13730] [Q&A] [answered]
I have put the RG15 into operation.
It works.
How can I configure the sensor via the tasmota console and mqtt? I don't know the right format.
Is there an example of the correct format to send cmnd's to Sensor90?
Thanks.

minniR2D2 (DE)

**Comment by sfromis** [comment, marked as answer]
Did you see the PDF document linked from docs for [Sensor90](https://tasmota.github.io/docs/Commands/#sensor90) ?

**Reply by minniR2D2** [reply]
Yes, the orders are known.
I'm looking for the complete structure of the string
for a MQTT Command Topic.
for example:   cmnd/RG15/**SENSOR = A** ..??, cmnd/RG15/**SENSOR A ..?**?, ??

**Reply by barbudor** [reply]
topic : cmnd/RG15/Sensor90
payload : command for Sensor90 as per doc/pdf

Don't forget the 90 attached to Sensor




















































There should be no = but that may depend on the tool that you are using for publishing MQTT
Apparently, except the command "B", no other command takes a parameter.
I would expect only 1 RG15 command per Sensor90 command
The best would be to test the commands on the WebConsole before trying through MQTT

**Reply by minniR2D2** [reply]
Thanks for the answers.
The cmds are correctly acknowledged in the tasmota console.
for example:
08:20:23.586 CMD: Sensor90 P
08:20:23.607 MQT: stat/RG15_RAIN/RESULT = {"SerialReceived":p"}
08:19:27.302 CMD: Sensor90 C
08:19:27.324 MQT: stat/RG15_RAIN/RESULT = {"SerialReceived":c"}
08:18:28.935 CMD: Sensor90 I
08:18:28.961 MQT: stat/RG15_RAIN/RESULT = {"SerialReceived":i"}
08:18:59.860 CMD: Sensor90 M
08:18:59.881 MQT: stat/RG15_RAIN/RESULT = {"SerialReceived":m"}

I noticed another problem (A):
In the console all values ​​are 0
tele/RG15_RAIN/SENSOR = {"Time":"2021-11-20T08:14:08","RG-15":{"Active":0.00, "Event":0.00, **"Total":0.00**, "FlowRate":0.00}}
The sensor sends, for example, "TotalAcc 1.82 mm".
the Tx-String: Acc  0.00 mm, EventAcc  0.00 mm, **TotalAcc  1.82** mm, RInt  0.00 mmph
The sensor is set to continuous mode (C).

I noticed another problem (B)!:
after a restart or power_off/on, the device settings are deleted!?!
before: 
D6 GPIO12	-> [HRG15 Rx]
D5 GPIO14	-> [HRG15 Tx]
after a restart:
D6 GPIO12	-> [None]
D5 GPIO14	-> [None]
Different pins, different ESP8266 the same problem.
That is my main problem now. I have to fix that first.
Then the other mqtt-problem.

**Reply by barbudor** [reply]
For problem B
- what is the result of 'savedata' command?
- configure, set 'Weblog 3', restart and provide the boot logs

Please please use code-marks (the <> button on top of the message box) to make your logs readable

Regarding point A, I have no idea yet on how the driver is working
How are you sure of what the RG is sending? Are you spying the serial link in parallel?

**Comment by barbudor** [comment]
What is the response to a command which is not a basic echo. For example 'Sensor90 A'?

**Reply by minniR2D2** [reply]
14:34:46.147 CMD: savedata
14:34:46.157 MQT: stat/RG15_RAIN/RESULT = {"SaveData":"ON"}

I have to solve the problem (B) first. Otherwise I won't get on with the other problems.
That won't work for a week. Then I'll be home again.
thank's for the Tipps
